plugins/microservices/src/get_server_property.cpp

```diff
@@ -0,0 +1,105 @@
+#include "irods/irods_server_properties.hpp"
+#include "irods/irods_ms_plugin.hpp"
+#include "irods/irods_re_structs.hpp"
+#include "irods/msParam.h"
+#include "irods/rodsErrorTable.h"
+#include "irods/irods_error.hpp"
+#include "irods/irods_logger.hpp"
+
+#include <exception>
+#include <functional>
+#include <string>
+
+namespace
+{
+    using log_msi = irods::experimental::log::microservice;
+
+    auto msi_impl(msParam_t* _in_name, msParam_t* _out_value, ruleExecInfo_t* rei) -> int
+    {
+        if (!rei || !rei->rsComm) {
+            log_msi::error("msi_get_server_property: input rei or rsComm is NULL.");
+            return SYS_INTERNAL_NULL_INPUT_ERR;
+        }
+
+        // ensure that this user is an admin
+        if (rei->uoic->authInfo.authFlag < LOCAL_PRIV_USER_AUTH) {
+            log_msi::error("msi_get_server_property: User {} is not local admin. status = {}",
+                           rei->uoic->userName,
+                           CAT_INSUFFICIENT_PRIVILEGE_LEVEL);
+            return CAT_INSUFFICIENT_PRIVILEGE_LEVEL;
+        }
+
+        try {
+            std::string name = parseMspForStr(_in_name);
+            const nlohmann::json* json_value = nullptr;
+            const auto& config = irods::server_properties::server_properties::instance().map();
```

Comment:
We have to be careful with `.map()` moving forward now that the server properties use locks.

`.map()` bypasses the locks and returns a reference to the underlying JSON object. Given that the agents are mostly single threaded, this is fine. However, we need to document the pitfalls of using `.map()` in an asynchronous context where the server properties are being updated.

Reply:
I'll add a note about that to the function in the irods_server_properties